Sources/Misc/SandboxEnvironmentDetector.swift

```diff
@@ -25,10 +25,14 @@ final class BundleSandboxEnvironmentDetector: SandboxEnvironmentDetector {
 
     private let bundle: Atomic<Bundle>
     private let isRunningInSimulator: Bool
+    private let receiptParser: PurchasesReceiptParser
 
-    init(bundle: Bundle = .main, isRunningInSimulator: Bool = SystemInfo.isRunningInSimulator) {
+    init(bundle: Bundle = .main,
+         isRunningInSimulator: Bool = SystemInfo.isRunningInSimulator,
+         receiptParser: PurchasesReceiptParser = PurchasesReceiptParser.default) {
```

Comment:
What do you think of injecting a protocol type so we don't tie this to receipt parser or fetcher (and make it easier to test)?